framework/wazuh/core/stats.py

```diff
@@ -219,59 +219,172 @@ def get_daemons_stats_(filename: str) -> list:
         raise WazuhInternalError(1121)
 
-    # Send message
-    s.send(command.encode())
-
-    # Receive response
     try:
-        rec_msg = s.receive().decode()
-    except ValueError:
-        raise WazuhInternalError(1118, extra_message="Data could not be received")
+        s.send(command.encode())
+        try:
+            return s.receive().decode()
+        except ValueError:
+            raise WazuhInternalError(1118, extra_message="Data could not be received")
+    finally:
+        s.close()
 
-    s.close()
 
-    # Format response
-    try:
-        data = json.loads(rec_msg)['data']
-        data.update((k, utils.get_utc_strptime(data[k], "%Y-%m-%d %H:%M:%S").strftime(common.DATE_FORMAT))
-                    for k, v in data.items() if k in {'last_keepalive', 'last_ack'})
-        return data
-    except Exception:
-        rec_msg = rec_msg.split(" ", 1)[1]
+def get_daemons_stats_from_socket(agent_id: str, daemon: str) -> dict:
+    """Get a daemon stats from an agent or manager.
+
+    Parameters
+    ----------
+    agent_id : str
+        Id of the agent to get stats from.
+    daemon : str
+        Name of the service to get stats from.
+
+    Returns
+    -------
+    dict
+        Object with daemon's stats.
+    """
+    if not agent_id or not daemon:
+        raise WazuhError(1307)
+
+    if not check_if_daemon_exists_in_agent(agent_id=agent_id, daemon=daemon):
+        raise WazuhError(1310)
+
+    dest_socket = get_stats_socket_path(agent_id=agent_id, daemon=daemon)
+    command = create_stats_command(agent_id=agent_id, daemon=daemon)
+
+    socket_msg = send_command_to_socket(dest_socket, command)
+    socket_response = json.loads(socket_msg)
+
+    # Handle error
+    if socket_response.get('error', 0) != 0:
+        rec_msg = socket_response.get('message', "")
         raise WazuhError(1117, extra_message=rec_msg)
+
+    data = socket_response['data']
+    data.update((k, utils.get_utc_strptime(data[k], "%Y-%m-%d %H:%M:%S").strftime(common.DATE_FORMAT))
+                for k, v in data.items() if k in {'last_keepalive', 'last_ack'})
+
+    while socket_response.get('remaining', False):
+        last_json_updated = socket_response.get('json_updated', False)
+        command = create_stats_command(agent_id=agent_id, daemon=daemon,
+                                       next_page=not last_json_updated)
+        rec_msg = send_command_to_socket(dest_socket, command)
+        socket_response = json.loads(rec_msg)
+
+        if last_json_updated:
+            data = socket_response['data']
+            data.update((k, utils.get_utc_strptime(data[k], "%Y-%m-%d %H:%M:%S").strftime(common.DATE_FORMAT))
+                        for k, v in data.items() if k in {'last_keepalive', 'last_ack'})
+        else:
+            page_data = socket_response['data']
+            # Extends existing items
+            [data[key].extend(item) for key, item in page_data.items() if key in data and isinstance(item, list)]
```

Comment:
This is wrong, the lists are nested in `data.global.files` but the `if` condition is checking whether `data.global` is a list (it is not a list but a dict). The same happens for `data.interval.files`. 

As a consequence, data is not extended and only partial results are being returned.